src/utilities.typ

```diff
@@ -114,6 +114,68 @@
     previous
 }
 
+// Convert a length of type length to pt.
+//
+// styles: from style()
+#let convert-length-type-to-pt(len, styles) = {
+    if "em" in repr(len) and styles == none {
+        panic("Cannot convert length to pt ('styles' not specified).")
+    }
+
+    // Convert the em part
+    //
+    // `len.em` is a float.
+    // We cannot simply do `len < 0pt` as that throws an error when
+    // `len` is compound (i.e., has both an absolute component and an em component).
+    // So, we have to handle the absolute and em components separately.
+    let em_len_in_pt = if len.em < 0 {
+        -measure(line(length: -len.em * 1em), styles).width
+    } else if len.em > 0 {
+        measure(line(length: len.em * 1em), styles).width
+    } else {
+        0pt
+    }
```

Comment:
1. By the way, I'm working on removing all usages of snake_case from the codebase, so I fully understand that you wanted this to be coherent with the rest :p. But let's use kebab-case for all future variable names so I can skip this later during my refactor.
2. We can't use `.em` as that was only added in Typst 0.7.0 iirc (tablex aims to support 0.2.0 or later, although increasing the minimum version later isn't really out of question, but I'd prefer to avoid it for tablex 0.1.0 for now). We'll probably have to perform some regex matching on repr() for now (a bit ugly, yeah)...

Reply:
>     2. We can't use `.em` as that was only added in Typst 0.7.0 iirc (tablex aims to support 0.2.0 or later, although increasing the minimum version later isn't really out of question, but I'd prefer to avoid it for tablex 0.1.0 for now). We'll probably have to perform some regex matching on repr() for now (a bit ugly, yeah)...  (Edit: same remark goes for the use of `.ratio` and `.length` below)

Ahh, I was wondering why you weren't using those. Makes sense. I'll make some changes.